src/Microsoft.DncEng.PatGeneratorTool/Program.cs

```diff
@@ -130,7 +132,16 @@ private static async Task<int> Go(List<AzureDevOpsPATScopes> scopes, string[] or
                     user = $"{user.Split('\\').Last()}@microsoft.com";
                 }
 
-                credentials = new VssAadCredential(user, password);
+                var app = PublicClientApplicationBuilder
+                    .Create(AzdoClientId)
+                    .WithAuthority("https://login.microsoftonline.com/microsoft.com/")
+                    .Build();
+
+                var passwordSecureStr = new SecureString();
+                Array.ForEach(password!.ToCharArray(), passwordSecureStr.AppendChar);
```

Comment:
This is 100% wrong usage of SecureString.  If the API that you're calling requires it, then you have to do it this way, of course.  If it just accepts a `System.String` then you shouldn't use SecureString.

Based on Chad's comment it sounds like this library is under development.  If so, please share with them that the docs for SecureString say not to use it.  (https://docs.microsoft.com/en-us/dotnet/api/system.security.securestring?view=net-6.0#remarks).  We (the BCL team) have decided that it _is_ a fully obsolete piece of tech, but we haven't quite marked it `[Obsolete]` yet because of legacy API impact.

Reply:
Oh: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2437 They literally just did this, and say it will get released in a few days.  Can we use that version and avoid all this SecureString stuff?